storage/transfermanager/downloader.go

```diff
@@ -366,6 +356,48 @@ func (d *Downloader) downloadWorker() {
 	d.workers.Done()
 }
 
+// startDownload downloads the first shard and schedules subsequent shards
+// if necessary.
+func (d *Downloader) startDownload(input *DownloadObjectInput) {
+	var out *DownloadOutput
+
+	// Special-case: full object read.
```

Comment:
I wouldn't really call this a special case as it is the most common case. Also, only reading partSize bytes is only relevant to large objects which must be sharded.

Reply:
True, I can remove this comment.